Build: Login

udenizdemirbilek opened the pull request:
Draft login page for review

Signs in test users with email&password authentication, redux is set up and working

To do:
Will add google and facebook login buttons,
Work on styling

udenizdemirbilek committed:
draft login1

udenizdemirbilek committed:
draft login1

udenizdemirbilek committed:
draft login2

udenizdemirbilek committed:
draft login3

halitbatur reviewed (commented):
@udenizdemirbilek so far so good

udenizdemirbilek committed:
Build: add login page and functionality

add login page and set up redux store to change state, pass userId to store,
navbar reads from store to render login, signup, profile and logout links conditionally

udenizdemirbilek commented:
add dispatcher to navbar to change state when user logs out, as well as selector to read state and conditionally render login, signup, logout and user profile buttons 
Please take another look @halitbatur

udenizdemirbilek committed:
login with navbar functionality

halitbatur commented on .vscode/launch.json:
```diff
@@ -0,0 +1,15 @@
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "pwa-chrome",
```
why is this file changed ?

udenizdemirbilek replied:
I think this file was missing and I tried to run vscode debugger a few days ago, and it created this file, I can either delete it or add it to gitignore

halitbatur commented on package.json:
```diff
@@ -3,18 +3,21 @@
         "bootstrap-icons": "^1.5.0",
-        "firebase": "^8.9.1",
+        "firebase": "8.9.1",
+        "firebaseui": "^5.0.0",
```
I don't think we need this package

udenizdemirbilek replied:
firebaseui removed!

halitbatur commented on src/components/Navbar/NavbarForLargeScreen.jsx:
```diff
@@ -1,9 +1,32 @@
+  const { userId, userName } = useSelector((state) => state.user);
+  const displayProfile = userId === "" ? "none" : "inline-block";
+  const displayLogin = userId === "" ? "inline-block" : "none";
+  const dispatch = useDispatch();
```
new line after this

halitbatur commented on src/containers/Login/Login.jsx:
```diff
@@ -0,0 +1,95 @@
+              value={login.email}
+            />
+          </Form.Group>
+
```
extra new line

halitbatur commented:
@udenizdemirbilek good work Deniz! I have written some changes, and u will need to update the branch and resolve conflicts

udenizdemirbilek committed:
final login

udenizdemirbilek committed:
final login

udenizdemirbilek committed:
final login2

ceyhungulbas reviewed (approved):
LGTM! 🎉
